opentracing/scope_manager.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2017 The OpenTracing Authors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+from .span import Span, SpanContext
+from .scope import Scope
+
+
+class ScopeManager(object):
+    """The `ScopeManager` interface abstracts both the activation of `Span`
+    instances (via `ScopeManager#activate(Span)`) and access to an active
+    `Span` / `Scope` (via `ScopeManager#active`).
+    """
+    def __init__(self):
+        # TODO: `tracer` should not be None, but we don't have a reference;
```

Comment:
this is just a nitpick and not really important to the discussion (so we can tackle even in another cleanup PR), but the idea was to have a `noop` module (or something similar) where we initialize all `noop_*` objects so we can use them everywhere.

Reply:
Yes, and I think we can leave it for another PR IHMO, so we decide on how to handle it cleanly (I tried to work on it myself, but there's no clear approach here, so I decided to leave it out for now).